addons/web/static/src/core/tree_editor/tree_editor_value_editors.js

```diff
@@ -322,6 +328,36 @@ function getPartialValueEditorInfo(fieldDef, operator, params = {}) {
             const options = fieldDef.selection || [];
             return makeSelectEditor(options, params);
         }
+        case "datetime_option": {
+            if (fieldDef.name in get_DATETIME_OPTIONS_WITH_SELECT()) {
+                const { options, defaultValue } = get_DATETIME_OPTIONS_WITH_SELECT()[fieldDef.name];
+                return { ...makeSelectEditor(options, params), defaultValue };
+            } else if (fieldDef.name === "__time") {
+                return {
+                    component: TimePicker,
+                    extractProps: ({ value, update }) => ({
+                        value: parseTime(value, true),
+                        onChange: (time) => update(time.toString(true)),
```

Comment:
we need to work with a fixed format (toString considers _is24HourFormat and _isMeridiemFormat)!